lib/msf/ui/console/command_dispatcher/modules.rb

```diff
@@ -387,7 +387,14 @@ def cmd_search(*args)
               end
             end
 
-            cached = true if args.empty?
+            if args.empty?
+              if @module_search_results.empty?
+                cmd_search_help
+                return false
+              end
+
+              cached = true
```

Comment:
I removed the redundant `else` in this nested conditional, since you're using early `return`.